infrastructure/cli/integration/mockserver.go

```diff
@@ -0,0 +1,54 @@
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"path"
+	"testing"
+
+	"goji.io"
+	"goji.io/pat"
+)
+
+type MockServer struct {
+	l    net.Listener
+	Host string
+	mux  *goji.Mux
+}
+
+func NewMockServer(t *testing.T) *MockServer {
```

Comment:
somewhat "common" testing pattern in go. You pass down `t *testing.T)` so you can do things like `t.Log` or `t.Fail` from your testing helpers